### ISSUE TITLE
Why are empty labels appearing in the redisgraph data?

### ISSUE THREAD

**Issue by p0xiao** [RedisGraph/RedisGraph#2690]
Hi @AviAvni @swilly22 @jeffreylovitz @gkorland @DvirDukhan , during use redisgraph(version 2.8.6), I found that empty labels appeared in redisgraph data, like this. Then I checked my code, the labels are all specified when the nodes are created.  Then I try to manually create nodes with empty labels, It is found that empty nodes cannot be created at all through "CREATE statement". Is this a bug ?  Or there will be other cases where empty nodes will be introduced?

<img width="655" alt="image" src="https://user-images.githubusercontent.com/10668702/199778790-97c9a8fc-46c5-4ab3-a3d0-d8e8bb6e3197.png">

<img width="643" alt="image" src="https://user-images.githubusercontent.com/10668702/199778629-9d2eb0b9-ac00-4135-8e2d-531efa2d2cfb.png">

RedisInsight export json file: 
<img width="350" alt="image" src="https://user-images.githubusercontent.com/10668702/199781004-a495c427-40f6-4e9b-81ed-d5ff85ee8114.png">

**Comment by LiorKogan** [comment]
2.8.6 was a pre-release version. It may well be due to a bug (several critical issues were fixed since that may explain this).
We suggest to upgrade to 2.8.20.

**Comment by p0xiao** [comment]
> First, please try the same with redis-cli or RedisInsight 2 to make sure it is not a RedisInsight 1 issue.
> 
> > I try to manually create nodes with empty labels, It is found that empty nodes cannot be created at all through "CREATE statement"
> 
> Do you get an error message when you do `graph.query g "create (x{a:1})"` ?
> 
> 2.8.6 was a pre-release version. It may well be due to a bug (several critical issues were fixed since which may explain this). We suggest upgrading to 2.8.20.

Thanks Reply.
I tried again to create empty node with your statement, it's successful.
```cypher
create (x{a:1})
```
The statement I used yesterday was `create (x:{a:1})`, more colons. And the creation statement in my code is like this ` "CREATE (a:%s {%s}) RETURN id(a)";`. So not caused by the creation.

Today, I try to restart redisgraph, failed to start...  It didn't start successfully until I cleared the data

redis.log
```
12538:M 04 Nov 2022 10:17:36.504 * <graph> Starting up RedisGraph version 2.8.6.
12538:M 04 Nov 2022 10:17:36.508 * <graph> Thread pool created, using 32 threads.
12538:M 04 Nov 2022 10:17:36.509 * <graph> Maximum number of OpenMP threads set to 32
12538:M 04 Nov 2022 10:17:36.509 * Module 'graph' loaded from /data/s/services/redisgraph/16479/server/so/redisgraph.so
12538:M 04 Nov 2022 10:17:36.509 * Loading RDB produced by version 6.0.16
12538:M 04 Nov 2022 10:17:36.510 * RDB age 1611 seconds
12538:M 04 Nov 2022 10:17:36.510 * RDB memory usage when created 658.54 Mb


=== REDIS BUG REPORT START: Cut & paste starting from here ===
12538:M 04 Nov 2022 10:17:37.292 # Redis 6.0.16 crashed by signal: 7, si_code: 128
12538:M 04 Nov 2022 10:17:37.292 # Crashed running the instruction at: 0x7f6c2afe0c95
12538:M 04 Nov 2022 10:17:37.292 # Accessing address: (nil)
12538:M 04 Nov 2022 10:17:37.292 # Failed assertion: <no assertion failed> (<no file>:0)

------ STACK TRACE ------
EIP:
/data/s/services/redisgraph/16479/server/so/redisgraph.so(RG_Matrix_setElement_UINT64+0x25)[0x7f6c2afe0c95]

Backtrace:
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479(logStackTrace+0x2e)[0x47a5ee]
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479(sigsegvHandler+0xa9)[0x47acb9]
/lib64/libpthread.so.0(+0xf5d0)[0x7f6c333255d0]
/data/s/services/redisgraph/16479/server/so/redisgraph.so(RG_Matrix_setElement_UINT64+0x25)[0x7f6c2afe0c95]
/data/s/services/redisgraph/16479/server/so/redisgraph.so(Graph_FormConnection+0x52)[0x7f6c2afdbeb2]
/data/s/services/redisgraph/16479/server/so/redisgraph.so(RdbLoadEdges_v9+0x94)[0x7f6c2afe3c34]
/data/s/services/redisgraph/16479/server/so/redisgraph.so(RdbLoadGraph_v9+0x2f3)[0x7f6c2afe4013]
/data/s/services/redisgraph/16479/server/so/redisgraph.so(+0x1ee185)[0x7f6c2afe2185]
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479[0x45b350]
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479(rdbLoadRio+0x248)[0x45d0e8]
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479(rdbLoad+0x4d)[0x45daad]
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479(loadDataFromDisk+0x8c)[0x43b5cc]
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479(main+0x470)[0x42d020]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7f6c32f6b3d5]
/data/s/services/redisgraph/16479/server/bin/redis-server *:16479[0x42d359]

------ INFO OUTPUT ------
# Server
redis_version:6.0.16
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:567eea098253f33e
redis_mode:standalone
os:Linux 3.10.0-957.el7.x86_64 x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:9.3.1
process_id:12538
run_id:556b6ace5aa85a92bf19000d49c6eca8fdc2f92d
tcp_port:16479
uptime_in_seconds:1
uptime_in_days:0
hz:50
configured_hz:50
lru_clock:6583872
executable:/data/s/services/redisgraph/16479/server/bin/redis-server
config_file:/data/s/services/redisgraph/16479/data/redis.conf
io_threads_active:0

# Clients
connected_clients:0
client_recent_max_input_buffer:0
client_recent_max_output_buffer:0
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0

# Memory
used_memory:62092776
used_memory_human:59.22M
used_memory_rss:0
used_memory_rss_human:0B
used_memory_peak:62092776
used_memory_peak_human:59.22M
used_memory_peak_perc:101.35%
used_memory_overhead:837952
used_memory_startup:837848
used_memory_dataset:61254824
used_memory_dataset_perc:100.00%
allocator_allocated:0
allocator_active:0
allocator_resident:0
total_system_memory:67383291904
total_system_memory_human:62.76G
used_memory_lua:37888
used_memory_lua_human:37.00K
used_memory_scripts:0
used_memory_scripts_human:0B
number_of_cached_scripts:0
maxmemory:4831838208
maxmemory_human:4.50G
maxmemory_policy:noeviction
allocator_frag_ratio:-nan
allocator_frag_bytes:0
allocator_rss_ratio:-nan
allocator_rss_bytes:0
rss_overhead_ratio:-nan
rss_overhead_bytes:0
mem_fragmentation_ratio:-nan
mem_fragmentation_bytes:0
mem_not_counted_for_evict:0
mem_replication_backlog:0
mem_clients_slaves:0
mem_clients_normal:0
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0

# Persistence
loading:1
rdb_changes_since_last_save:0
rdb_bgsave_in_progress:0
rdb_last_save_time:1667528256
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0
module_fork_in_progress:0
module_fork_last_cow_size:0
loading_start_time:1667528256
loading_total_bytes:12925073
loading_loaded_bytes:12582911
loading_loaded_perc:97.35
loading_eta_seconds:0

# Stats
total_connections_received:0
total_commands_processed:0
instantaneous_ops_per_sec:0
total_net_input_bytes:0
total_net_output_bytes:0
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:0
evicted_keys:0
keyspace_hits:0
keyspace_misses:0
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0
total_reads_processed:0
total_writes_processed:0
io_threaded_reads_processed:0
io_threaded_writes_processed:0

# Replication
role:master
connected_slaves:0
master_replid:c803a9bff395496178abf73b64b45714e6fbeb34
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:536870912
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:0.063492
used_cpu_user:0.739735
used_cpu_sys_children:0.000000
used_cpu_user_children:0.000000

# Modules
module:name=graph,ver=20806,api=1,filters=0,usedby=[],using=[],options=[]

# Commandstats

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=1,expires=0,avg_ttl=0

------ CLIENT LIST OUTPUT ------

------ REGISTERS ------
12538:M 04 Nov 2022 10:17:37.295 # 
RAX:0000000000000006 RBX:00000000000001db
RCX:00000000000001d6 RDX:00000000000001d6
RDI:64757469676e6f6c RSI:00000000000083fd
RBP:64757469676e6f6c RSP:00007fffc2822c50
R8 :0000000000000000 R9 :0000000000008000
R10:00007f6c2b0eb68c R11:0000000000000000
R12:00000000000083fd R13:00000000000001d6
R14:64757469676e6f6c R15:00000000000083fc
RIP:00007f6c2afe0c95 EFL:0000000000010206
CSGSFS:0000000000000033
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c5f) -> 00007fffc2822ed0
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c5e) -> 00007fffc2822ed0
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c5d) -> 00007f6c2afdbeb2
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c5c) -> 00000000000083fc
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c5b) -> 64757469676e6f6c
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c5a) -> 00000000000001d6
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c59) -> 00000000000083fd
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c58) -> 00007f6c32a351e0
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c57) -> 00000000000001db
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c56) -> 0000000000000001
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c55) -> 00007f6c2aff99f1
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c54) -> 00007fffc2822d30
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c53) -> 00007f6c32a28000
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c52) -> 0000000000000001
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c51) -> 00007f6c2afe04b2
12538:M 04 Nov 2022 10:17:37.295 # (00007fffc2822c50) -> 00000000000001d6

------ MODULES INFO OUTPUT ------
# graph_executing commands

------ FAST MEMORY TEST ------
12538:M 04 Nov 2022 10:17:37.296 # Bio thread for job type #0 terminated
12538:M 04 Nov 2022 10:17:37.296 # Bio thread for job type #1 terminated
12538:M 04 Nov 2022 10:17:37.296 # Bio thread for job type #2 terminated
*** Preparing to test memory region 5c9000 (2277376 bytes)
*** Preparing to test memory region 1005000 (135168 bytes)
*** Preparing to test memory region 7f6c136a7000 (63963136 bytes)
*** Preparing to test memory region 7f6c173a8000 (8388608 bytes)
*** Preparing to test memory region 7f6c17ba9000 (8388608 bytes)
*** Preparing to test memory region 7f6c183aa000 (8388608 bytes)
*** Preparing to test memory region 7f6c18bab000 (8388608 bytes)
*** Preparing to test memory region 7f6c193ac000 (8388608 bytes)
*** Preparing to test memory region 7f6c19bad000 (8388608 bytes)
*** Preparing to test memory region 7f6c1a3ae000 (8388608 bytes)
*** Preparing to test memory region 7f6c1abaf000 (8388608 bytes)
*** Preparing to test memory region 7f6c1b3b0000 (8388608 bytes)
*** Preparing to test memory region 7f6c1bbb1000 (8388608 bytes)
*** Preparing to test memory region 7f6c1c3b2000 (8388608 bytes)
*** Preparing to test memory region 7f6c1cbb3000 (8388608 bytes)
*** Preparing to test memory region 7f6c1d3b4000 (8388608 bytes)
*** Preparing to test memory region 7f6c1dbb5000 (8388608 bytes)
*** Preparing to test memory region 7f6c1e3b6000 (8388608 bytes)
*** Preparing to test memory region 7f6c1ebb7000 (8388608 bytes)
*** Preparing to test memory region 7f6c1f3b8000 (8388608 bytes)
*** Preparing to test memory region 7f6c1fbb9000 (8388608 bytes)
*** Preparing to test memory region 7f6c203ba000 (8388608 bytes)
*** Preparing to test memory region 7f6c20bbb000 (8388608 bytes)
*** Preparing to test memory region 7f6c213bc000 (8388608 bytes)
*** Preparing to test memory region 7f6c21bbd000 (8388608 bytes)
*** Preparing to test memory region 7f6c223be000 (8388608 bytes)
*** Preparing to test memory region 7f6c22bbf000 (8388608 bytes)
*** Preparing to test memory region 7f6c233c0000 (8388608 bytes)
*** Preparing to test memory region 7f6c23bc1000 (8388608 bytes)
*** Preparing to test memory region 7f6c243c2000 (8388608 bytes)
*** Preparing to test memory region 7f6c24bc3000 (8388608 bytes)
*** Preparing to test memory region 7f6c253c4000 (8388608 bytes)
*** Preparing to test memory region 7f6c25bc5000 (8388608 bytes)
*** Preparing to test memory region 7f6c263c6000 (8388608 bytes)
*** Preparing to test memory region 7f6c26bc7000 (8388608 bytes)
*** Preparing to test memory region 7f6c273c8000 (8388608 bytes)
*** Preparing to test memory region 7f6c27bc9000 (8388608 bytes)
*** Preparing to test memory region 7f6c283ca000 (8388608 bytes)
*** Preparing to test memory region 7f6c28bcb000 (8388608 bytes)
*** Preparing to test memory region 7f6c293cc000 (8388608 bytes)
*** Preparing to test memory region 7f6c29bcd000 (8388608 bytes)
*** Preparing to test memory region 7f6c2a3ce000 (8388608 bytes)
*** Preparing to test memory region 7f6c2c0d2000 (16384 bytes)
*** Preparing to test memory region 7f6c32600000 (8388608 bytes)
*** Preparing to test memory region 7f6c33311000 (20480 bytes)
*** Preparing to test memory region 7f6c3352e000 (16384 bytes)
*** Preparing to test memory region 7f6c33e4f000 (20480 bytes)
*** Preparing to test memory region 7f6c33e5e000 (8192 bytes)
*** Preparing to test memory region 7f6c33e60000 (4096 bytes)
*** Preparing to test memory region 7f6c33e63000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O
Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.

------ DUMPING CODE AROUND EIP ------
Symbol: RG_Matrix_setElement_UINT64 (base: 0x7f6c2afe0c70)
Module: /data/s/services/redisgraph/16479/server/so/redisgraph.so (base 0x7f6c2adf4000)
$ xxd -r -p /tmp/dump.hex /tmp/dump.bin
$ objdump --adjust-vma=0x7f6c2afe0c70 -D -b binary -m i386:x86-64 /tmp/dump.bin
------
12538:M 04 Nov 2022 10:17:39.264 # dump of function (hexdump of 165 bytes):
41574989f7415641554989cd41544989d44889ca554c89e64889fd534883ec38e8bb010000488b7d204885ff74124c89e24c89eee887f7ffff4189c685c07564488b5d18488b4510488d7c24184c89e94c89e248893c244c8b75084889de4889442408e81886030085c075544c89ea4c89e64889dfe846e403004c89f74c89e94c89e24c89fee8f5f803004189c64889ef4c89ea4c89e6e8640100004889efe8cc0c000048

=== REDIS BUG REPORT END. Make sure to include from START to END. ===

       Please report the crash by opening an issue on github:

           http://github.com/redis/redis/issues

  Suspect RAM error? Use redis-server --test-memory to verify it.
```

**Comment by LiorKogan** [comment]
@p0xiao 
Thank you for submitting this crash log.
The root cause, with very high probability, is an issue that was solved in 2.8.14
